scripts/automation/dismod/build_singularity.sh

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bash
```

Comment:
This script is a hacked copy of the build_singularity.sh that Grant made. He may have updated his since this was copied. I can't recall why I hacked it, either. In short, we need to go back and try Grant's latest version, using his instructions on the hub.

Reply:
I have a copy of Grant's latest version.  There are some lines that show up in the diff, that look the same to me; for example, one example is:
41c41
<   fi
---
>   fi

There are also two places where I can see differences in the text:

1)  In our copy of build_singularity.sh, "dev" below is "devnotgonnahappen", is there a good reason for that?

## Set CPUs to pin to using numactl -- for dev cluster, CPU enforcement is enabled so we can pull the CPUs directly. For cluster-prod, need
 to generate a list
    if [ $SGE_CLUSTER_NAME == "dev" ]
      then
        cpu_list="$(cat /proc/$$/status | grep '^Cpus_allowed_list:' | awk '{print $NF}')"
      else
		target_thread=$(($threads - 1))
        cpu_list="0-$target_thread"
      fi

2) In our copy of build_singularity.sh,  we write "singularity" in the line below,

numactl -C +$cpu_list singularity build $file_path docker://reg.ihme.uw.edu/$image_name

as: 
/share/local/singularity-2.4.2/bin/singularity

is there a good reason for that?

Reply:
I think the change to just "singularity" means the path is updated for everybody. The part about devnotgonnahappen is that the allowed CPUS list is a good idea but not good in practice, so I don't want to do it. You are pinning processes to CPUs, but other processes won't play along, so it won't work.